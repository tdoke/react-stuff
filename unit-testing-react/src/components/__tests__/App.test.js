import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import TestComponent from '../App';

const mockStore = configureStore([]);


describe('TestComponent', () => {
  let props;
  let getComponent;
  let store;


  beforeEach(() => {
    props = {};
    store = mockStore({
      comments: []
    });
    getComponent = () => render(<Provider store={store}>
      <TestComponent {...props} />
      </Provider>);
  });

  describe('render', () => {
    
    it('TestComponent should render with empty list', () => {
      const { getByTestId } = getComponent();
      expect(getByTestId('commentsList')).toMatchSnapshot();
    });

    it('TestComponent should render with one list item', async () => {
      const { getByTestId } = getComponent();
      const input = getByTestId('comment-value');
      const submitBtn = getByTestId('button-submit');
      fireEvent.change(input, {target: {value: 'test-comment'}});
      fireEvent.click(submitBtn);
      await waitFor(() => screen.getByTestId('commentsList'));
      expect(getByTestId('commentsList')).toMatchSnapshot();
    });
  });
});
