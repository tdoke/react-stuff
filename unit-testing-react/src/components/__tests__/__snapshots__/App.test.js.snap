// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent render TestComponent should render with empty list 1`] = `
<ul
  data-testid="commentsList"
/>
`;

exports[`TestComponent render TestComponent should render with one list item 1`] = `
<ul
  data-testid="commentsList"
/>
`;
